chef/templates/default/sysvinit.sh.erb

```diff
@@ -33,32 +33,42 @@ _reload() {
   start-stop-daemon --stop --quiet --pidfile "<%= @pid_file %>" --user "<%= @user %>" --signal "<%= @reload_signal %>"
 }
 
-<%- elsif @platform_family == 'rhel' -%>
-. /etc/rc.d/init.d/functions
-
-_pid () {
-  pidof -s -o $$ -o $PPID -o %PPID -x "$1" || pidof -s -o $$ -o $PPID -o %PPID -x "${1##*/}"
-}
-
+<%- else -%>
 _start() {
-  ( cd "<%= @directory %>" && daemon --user "<%= @user %>" --pidfile "<%= @pid_file %>" "<%= @daemon %> <%= @daemon_options %>" >/dev/null 2>&1 ) &
-<%- unless @pid_file_external -%>
-  sleep 1 # Give it some time to start before checking for a pid
-  _pid "<%= @daemon %>" > "<%= @pid_file %>" || return 3
-<%- end -%>
-  return 0 <%# ¯\_(ツ)_/¯ %>
+  <%# Implementing this using RedHat's bash helpers is too painful. Sorry. %>
+  <%# See dummy.rb for a more commented version of this code. %>
+  /opt/chef/embedded/bin/ruby <<EOH
+pid_file = <%= @pid_file.inspect %>
+File.unlink(pid_file) if File.exist?(pid_file)
```

Comment:
My mind is a few too levels in shell, template and ruby... but does this mean that this isn't idempotent?:
```sh
% /etc/init.d/foosvc start
% /etc/init.d/foosvc start
```